prime_factor/prime_factor.c

```diff
@@ -0,0 +1,37 @@
+#include <stdio.h>
+
+void generate_primes(int *ar, int *cnt) {
+    *cnt = 0;
+    for (int i = 2; i < 60000 && *cnt < 1000; ++i) {
```

Comment:
Can we get rid of the constants like `60000` and `1000` (and many other instance in program)?